### PR TITLE
Merge front_responsive to master

### LINKED ISSUE #7: Changes needed in the frontend
# Description
The frontend needs to be responsive to ensure a good user experience.
FormLab has to be rewrited to use react-hook-form

# Objectives
- Disable animations on News component
- add media queries for mobile
- turn navBar into a menu
- rewrite formLab with react-hook-form
All changes will be made in a branch front_responsive.

### PR DESCRIPTION
**Pull request by MatiSalinas**
#### What does this PR do?
This pull request addresses the responsiveness of the frontend and refactors the FormLab component to use `react-hook-form` for better form management. It includes the following changes:

- Disables animations on the News component for mobile devices.
- Adds media queries for devices with a screen width of 800px or less, optimizing the layout for smaller screens.
- Transforms the navbar into a mobile-friendly menu using the `isMobile` flag to provide different layouts for mobile and desktop views.
- Rewrites the FormLab component to use `react-hook-form` for handling form data and integrates `zod` for form validation.

#### Changes Made:
1. **Animations in News Component:**
   - Introduced a flag (`isMobile`) to disable `framer-motion` animations on mobile devices, improving the user experience on smaller screens.
  
2. **Mobile Responsiveness:**
   - Added media queries targeting screens 800px and below.
   - Adjusted the layout by switching from `flex` to `column` for most elements and setting component widths to `100vw`.

3. **Navbar Refactor:**
   - Used the `isMobile` flag to provide a responsive menu, replacing the traditional navbar with a collapsible menu on mobile devices.

4. **FormLab Refactor:**
   - Replaced the form state management with `react-hook-form`'s `register` method.
   - Integrated `zod` for schema-based validation, ensuring consistent validation and error handling.

#### Related Issues:
- [Link to related issue #7](https://github.com/Esc-IngComp-UNC/computer-laboratory-web-page/issues/7)

### CODE REVIEW

**Comment by prietojulii** [comment]
Great job, Mat! I only have a few comments

File: web/src/App.css [hunk 1]
```diff
@@ -0,0 +1,679 @@
...
+  flex-direction: column;
+  justify-content: center;
+  
+}
+nav{
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  color: #017481;
+  background-color: #f2f3f4;
```
Comment (prietojulii) [inline web/src/App.css:25]: We can use CSS variables to make maintenance easier. For example, colors could be placed in variables and reused

File: web/src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import Contact from '@components/Contact'
+import { Element } from 'react-scroll'
+
+function App() {
+
+  return (
+      <div className="container">
+        <NavBar></NavBar>
+        <CarouselImages></CarouselImages>
+        <Element name="aboutUs">
```
Comment (prietojulii) [inline web/src/App.jsx:18]: The words displayed to the user must be in Spanish

File: web/src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        <AboutUs></AboutUs>
+        </Element>
+        <Element name="novedades">
+        <News></News>
+        </Element>
+        <Element name="faq">
```
Comment (prietojulii) [inline web/src/App.jsx:24]: the same here

File: web/src/components/Contact.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import youtube from '/svgs/youtube.svg'
+import instagram from '/svgs/instagram.svg'
+import linkedin from '/svgs/linkedin.svg'
+import twitter from '/svgs/twitterx.svg'
+
+function Contact(){
+    //TODO add SVG to each link
```
Comment (prietojulii) [inline web/src/components/Contact.jsx:7]: This comment can be deleted?

File: web/.gitkeep
Comment (prietojulii) [inline web/.gitkeep:1]: this file can be deleted

File: web/index.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<!doctype html>
+<html lang="es">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Laboratorio Computacion</title>
```
Comment (prietojulii) [inline web/index.html:6]:
```suggestion
    <title>Laboratorio de Computación</title>
```

File: web/src/components/Contact.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import youtube from '/svgs/youtube.svg'
+import instagram from '/svgs/instagram.svg'
+import linkedin from '/svgs/linkedin.svg'
+import twitter from '/svgs/twitterx.svg'
```
Comment (prietojulii) [inline web/src/components/Contact.jsx:4]: We havent twitter or youtube. Can you delete them?

File: web/src/components/Contact.jsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+function Contact(){
+    //TODO add SVG to each link
+    return (
+        <div className="contactContainer">
+            <div className="infoL">
+            <p>Laboratorio de Computación:</p>
+
+            <p>Correo: lc.fcefyn@gmail.com</p>
+            <p>Instagram: c.fcefyn</p>
```
Comment (prietojulii) [inline web/src/components/Contact.jsx:14]:
```suggestion
            <div className="socialMedia">
                <img src={instagram} alt="InstagramLogo" />
            <a target='_blank' href='https://www.instagram.com/lc.fcefyn/'>Instagram</a>
            </div>
```

File: web/src/components/Contact.jsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import twitter from '/svgs/twitterx.svg'
+
+function Contact(){
+    //TODO add SVG to each link
+    return (
+        <div className="contactContainer">
+            <div className="infoL">
+            <p>Laboratorio de Computación:</p>
+
+            <p>Correo: lc.fcefyn@gmail.com</p>
```
Comment (prietojulii) [inline web/src/components/Contact.jsx:13]: Can we use an image/icon for GMAIL. For example https://upload.wikimedia.org/wikipedia/commons/7/7e/Gmail_icon_%282020%29.svg

File: web/src/components/MapGoogle.jsx [hunk 9]
```diff
@@ -0,0 +1,12 @@
+function MapGoogle(){
+    return(
+    <div className="mapa" >
+        <div className="borderT"></div>
+
+        <div className="mapa"><iframe src="https://www.google.com/maps/embed?pb=!1m10!1m8!1m3!1d13617.083057234764!2d-64.181348!3d-31.434212!3m2!1i1024!2i768!4f13.1!5e0!3m2!1ses-419!2sar!4v1727935597767!5m2!1ses-419!2sar" width={'100%'}  height={600} style={{border:0}} allowFullScreen={''} loading={"lazy"} referrerPolicy={"no-referrer-when-downgrade"}></iframe></div>
```
Comment (prietojulii) [inline web/src/components/MapGoogle.jsx:6]:
```suggestion
        <div className="mapa"><iframe src="https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d13617.083057234764!2d-64.181348!3d-31.434212!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x9432a2f54d372bf1%3A0xf1d83ddc363cda16!2sFacultad%20de%20Ciencias%20Exactas%2C%20F%C3%ADsicas%20y%20Naturales!5e0!3m2!1ses-419!2sar!4v1731029204729!5m2!1ses-419!2sar" width={'100%'}  height={600} style={{border:0}} allowFullScreen={''} loading={"lazy"} referrerPolicy={"no-referrer-when-downgrade"}></iframe></div>
```

Comment (prietojulii) [reply web/src/components/MapGoogle.jsx:6]: Esto marca un punto rojo en la ubicación

File: web/src/App.jsx (hunk 2 above)
Comment (prietojulii) [inline web/src/App.jsx:18]: Maintain the order of the menu with the presentation order.
1- aboutus
2- novedades
3- faq
4-contacto

File: web/src/components/Contact.jsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+            <p>Instagram: c.fcefyn</p>
+            </div>
+            <div className="infoR">
+            <p>FCEFYN:</p>
+
+            <div className="socialMedia">
+                <img src={twitter} alt="twitterXLogo" />    
+            <a target='_blank' href='https://twitter.com/FCEFyN_UNC' >twitter</a>
```
Comment (prietojulii) [inline web/src/components/Contact.jsx:21]:
```suggestion
            <a target='_blank' href='https://twitter.com/fcefyn_unc' >twitter</a>
```

File: web/src/components/Contact.jsx (hunk 4 above)
Comment (prietojulii) [inline web/src/components/Contact.jsx:7]:
```suggestion
```

**Comment by MatiSalinas** [comment]
Thanks for your feedback, I made the suggested changes in the last commit [0aa27ba](https://github.com/Esc-IngComp-UNC/computer-laboratory-web-page/pull/8/commits/0aa27ba183043bea8ecdc585a149fdea1c351891).

**Comment by prietojulii** [comment]
It's amazing! Good Job!

**Review by prietojulii — APPROVED** [review]

**Review by prietojulii — APPROVED** [review]

File: web/src/main.jsx [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import { StrictMode } from 'react'
+import { createRoot } from 'react-dom/client'
+import App from './App.jsx'
```
Comment (GabrielEValenzuela) [inline web/src/main.jsx:3]:
```suggestion
import App from '@/src/App.jsx'
```

**Review by GabrielEValenzuela — CHANGES_REQUESTED** [review]
Great work Mati! 

Please check missing endlines and one little improvement on the path

**Comment by MatiSalinas** [comment]
added endline and changed path on [27635ab](https://github.com/Esc-IngComp-UNC/computer-laboratory-web-page/commit/27635aba63d92699104e06b16fba4c6528c3aa9f).

**Review by GabrielEValenzuela — APPROVED** [review]
LGTM

**Pull request merged**